frontend/src/components/Final/Final.tsx

```diff
@@ -13,101 +13,126 @@ import FinalButton from "./FinalButton";
+  onNext: () => void;
 }
 
 /**
  * Final is a block view that shows the final scores of the block
  * It can only be the last view of a block
  */
 const Final = ({
-    block,
-    participant,
-    score,
-    final_text,
-    action_texts,
-    button,
-    onNext,
-    show_participant_link,
-    participant_id_only,
-    show_profile_link,
-    social,
-    feedback_info,
-    points,
-    rank,
-    logo,
-    percentile,
+  block,
+  participant,
+  score,
+  final_text,
+  action_texts,
+  button,
+  onNext,
+  show_participant_link,
+  participant_id_only,
+  show_profile_link,
+  social,
+  feedback_info,
+  points,
+  rank,
+  logo,
+  percentile,
 }: FinalProps) => {
 
-    const session = useBoundStore((state) => state.session);
-    const navigate = useNavigate();
-
-    useEffect(() => {
-        finalizeSession({ session: session!, participant });
-    }, [session, participant]);
-
-    return (
-        <div className="aha__final d-flex flex-column justify-content-center">
-            {rank && (
-                <div className="text-center">
-                    <Rank cup={{ className: rank?.class, text: rank.text }} score={{ score, label: points }} />
-                </div>
-            )}
-            {(typeof percentile === 'number' && percentile >= 0 && percentile <= 100) && (
-                <div className={classNames("aha__final-rank-bar-section", rank?.class)}>
-                    <div data-testid="final-rank-bar-cursor" className="aha__final-rank-bar-cursor" style={{ left: `${percentile}%` }}></div>
-                </div>
-            )}
-            <div className="aha__final-text">
-                <div dangerouslySetInnerHTML={{ __html: final_text }} />
+  const session = useBoundStore((state) => state.session);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    finalizeSession({ session: session!, participant });
+  }, [session, participant]);
+
+  // To be implemented!
+  const totalScore = 1234;
+  
+  const hasPercentile = typeof percentile === 'number' && percentile >= 0 && percentile <= 100;
+  const timeline = getTimeline(["dot", "dot", "star-4", "dot", "dot", "star-5", "dot", "dot", "star-6", "dot", "dot", "star-7"])
+  const step = 2
+
+  return (
+    <div className="aha__final p-3 d-flex flex-column justify-content-center pt-5 mx-auto mt-5"  style={{maxWidth: "500px"}}>
+
+      <ScoreBoard
+        score={score}
+        totalScore={totalScore}
+        percentile={percentile}
+        timeline={timeline}
+        step={7} />
+
+        {/* Medal */}
+        {/* <div class="list-group-item">
+          {rank && <div className="text-center">
+            <Rank cup={{ className: rank?.class, text: rank.text }} score={{ score, label: points }} />
+          </div>}
+        </div> */}
+        
+        {/* Also contains the statistics */}
+        {/* <div className="aha__final-text">
+              <div dangerouslySetInnerHTML={{ __html: final_text }} />
+          </div> */}
```

Comment:
Commented out code is not a good idea in general, it makes it very hard to understand what has been changed.